### ISSUE TITLE
Speed up image_fitting_with_sherpa tutorial

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#2186]
Currently `make docs-all` takes about 10 minutes. I think we should try to speed it up, and one way to do that is to bring the execution time of the slowest tutorial notebooks down.

This issue is to discuss https://docs.gammapy.org/0.11/notebooks/image_fitting_with_sherpa.html since
```
INFO:gammapy.scripts.jupyter:   ... EXECUTING temp/image_fitting_with_sherpa.ipynb
INFO:gammapy.scripts.jupyter:   ... Executing duration: 104.3 seconds
```

Most of the time is spent here:
https://docs.gammapy.org/0.11/notebooks/image_fitting_with_sherpa.html#Iteratively-find-and-fit-additional-sources

Suggest to make some change there, e.g. to change optimiser to levmar, or some fit tol parameter, so that it runs in < 10 sec and ~ 100 likelihood evaluations, not 1 minute and ~ 1000 likelihood evaluations.

The code there isn't great anyways, can be cleaned up a bit. Currently 10 Gaussian model components are made, but then only 3 used.  Also I think:
```
gs = [g0, g1, g2]
for i in range(1, len(gs)):
```
can be replaced with a bit clearer variable names and the `i` isn't needed, i.e.
```
sources = ...
for source in sources:
```

@registerrier or @AtreyeeS - OK to do this change? Would you be willing to send a PR?

**Comment by AtreyeeS** [comment]
I agree! The tutorials do take a lot of time to run. I will put in a PR later this week.

**Comment by cdeil** [comment]
Fixed by @AtreyeeS in #2203 .
